backend/src/main/java/com/woowacourse/thankoo/coupon/domain/Coupons.java

```diff
@@ -24,4 +42,38 @@ public List<Long> getReceiverIds() {
                 .map(Coupon::getReceiverId)
                 .collect(Collectors.toList());
     }
+
+    public Long getRepresentativeSenderId() {
+        Long representativeSenderId = values.get(REPRESENTATIVE_INDEX).getSenderId();
+        validateSameSender(representativeSenderId);
+        return representativeSenderId;
+    }
+
+    private void validateSameSender(final Long representativeSenderId) {
+        if (isNotSameSender(representativeSenderId)) {
+            throw new InvalidCouponException(ErrorType.NOT_IN_SAME_COUPON_GROUP);
+        }
+    }
+
+    private boolean isNotSameSender(final Long representativeSenderId) {
+        return values.stream()
+                .anyMatch(value -> !value.isSender(representativeSenderId));
+    }
+
```

Comment:
```suggestion
    private void validateSameSender(final Long representativeSenderId) {
        if (!hasSameSenders(representativeSenderId)) {
            throw new InvalidCouponException(ErrorType.NOT_IN_SAME_COUPON_GROUP);
        }
    }

    private boolean hasSameSenders(final Long representativeSenderId) {
        return values.stream()
                .allMatch(value -> value.isSender(representativeSenderId));
    }
```

부정문의 메서드에 부정으로 매칭을 하고 있어서 살묘시 제안해봅니당. CouponContent 도 마찬가지에요!!

Reply:
오 감사합니다 ㅎㅎ